dashboard/app/models/level.rb

```diff
@@ -349,6 +351,27 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  def log_changes
+    log = JSON.parse(audit_log || "[]")
+
+    # serialize 'properties' changes
+    changed = self.changed
+    if changed.include?('properties')
+      changed.delete('properties')
+      changed_attributes['properties'].each do |key, value|
+        changed.push(key) if properties[key] != value
+      end
+    end
+
+    log.push(
+      {
+        changed_at: Time.now,
+        changed: changed
```

Comment:
It is not, because the rails model (appropriately) has no concept of user auth; that stuff belongs in the controller.

It would sure be nice to have something like that in here and there are a couple of ways we could go about doing it, but I'm going for quick 'n' easy.